.nginx/nginx.conf

```diff
@@ -0,0 +1,15 @@
+worker_processes 4;
+
+events { worker_connections 1024; }
```

Comment:
Looking through more documentation, I now understand that `max clients = worker_processes * worker_connections`, which shows how many users we can handle, right? Do you think you can give a little more context for what exactly a worker_process and worker_connection is?

Reply:
The worker connections is the number of connections that an individual process can handle. By default it is set to 1024 which I didn't change but its possible to vary the number depending on what Nginx is really doing to tune performance. What it means to have 1024 connections is that one process can concurrently handle that many clients. A client represents a network request (for example someone opening up the website). So one process can handle 1024 people accessing the website at once and serve them the necessary files to render.

A worker process is what nginx spawns to actually perform the job of serving files (html in this case cuz its a website). Ideally as mentioned previously we should set it to `auto` so that nginx can optimize by itself based on the configuration of the machine that it is running on. Usually the best number to set it to would be the number of cores running to ensure that nginx can best schedule tasks between them. If say we only have one core but 2 worker processes then each would then get ~50% of the resources offered by that one cpu. In reality its not an exact 50/50 split because context switching consumes resources as well. Typically for things like running websites its recommended to scale up the number of connections (because each connection really doesn't do much, just renders and html file) when demand goes up. 

Note: This all deals with optimizing when trying to scale and I very much doubt we'll have >1000 concurrent users but its always good to be aware of how a scalable architecture is setup.